xunit-autolabeler-v2/ast_parser/python/source_parser.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import ast
+import os
+
+from .source_parsers import direct_invocation, webapp2_router, flask_router
+from typing import Any, List
+
+
+def _get_method_children(expr: Any) -> List[Any]:
+    results = []
+
+    if hasattr(expr, 'id'):
+        results.append(expr.id)  # Base case
+
+    if hasattr(expr, 'body') and isinstance(expr.body, list):
+        for sub_expr in expr.body:
+            results += _get_method_children(sub_expr)
+
+    if hasattr(expr, 'value'):
+        results += _get_method_children(expr.value)
+
+    if hasattr(expr, 'func'):
+        func_list = expr.func
+        if not isinstance(func_list, list):
+            # Not all func values are lists!
+            func_list = [func_list]
+
+        for func in func_list:
+            results += _get_method_children(func)
+            if hasattr(func, 'args'):
+                for arg in func.args:
+                    results += _get_method_children(arg)
+
+    return results
+
+
+def _get_ending_line(expr: Any) -> int:
+    m_end_stmt = expr
```

Comment:
What does this stand for, in general names should be descriptive, avoid abbreviation.

Reply:
Renamed.